products/BellHybrid/apps/application-bell-meditation-timer/presenter/SettingsPresenter.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "SettingsPresenter.hpp"
+
+#include "models/ChimeInterval.hpp"
+#include "models/ChimeVolume.hpp"
+#include "models/StartDelay.hpp"
+
+#include <common/widgets/list_items/Fraction.hpp>
+#include <common/widgets/list_items/Number.hpp>
+#include <common/LanguageUtils.hpp>
+#include <apps-common/InternalModel.hpp>
+#include <apps-common/widgets/spinners/Spinners.hpp>
+#include <widgets/ListViewEngine.hpp>
+
+namespace app::list_items
+{
+    struct StartDelayFormatter
+    {
+      public:
+        StartDelayFormatter() : none(utils::translate("app_bell_meditation_interval_none"))
+        {}
+        std::string operator()(std::uint32_t value) const
+        {
+            return value == 0 ? none : std::to_string(value);
+        }
+
+      private:
+        std::string none;
+    };
+
+    using StartDelaySpinner = gui::UIntegerSpinnerWithFormatter<StartDelayFormatter>;
+    class StartDelay : public app::list_items::details::ListItemBase<StartDelaySpinner>
+    {
+      public:
+        StartDelay(spinner_type::range &&range,
+                   gui::AbstractSettingsModel<std::uint32_t> &model,
+                   const std::string &topDescription    = "",
+                   const std::string &bottomDescription = "")
+            : app::list_items::details::ListItemBase<spinner_type>(
+                  std::move(range), model, topDescription, bottomDescription)
+        {}
+
+      private:
+        void control_bottom_description(const std::uint32_t &value) final
+        {
+            bottomText->setVisible(value != 0);
+            bottomText->setRichText(utils::language::getCorrectSecondsNumeralForm(value));
+        }
+    };
+
+} // namespace app::list_items
+
+namespace app::meditation
+{
+    using namespace gui;
+    SettingsPresenter::SettingsPresenter(app::ApplicationCommon *app,
+                                         models::ChimeInterval &chimeIntervalModel,
```

Comment:
minor: all references probably could be const

Reply:
It won't work as each provider's item needs to have access to both `setValue` and `getValue` :(